### PR TITLE
FEI-4957.1: Remove file extensions from imports/requires

### PR DESCRIPTION
**Pull request by kevinbarabash**
## Summary:
TypeScript doesn't work with file extensions in imports.  This PR removes all .js file imports and requires.

I use find'n'replace with `.js";` and `.js");`.  This was imprecise and require a bunch of manual fixes.  In the future I will try using eslint's auto-fixing and if that doesn't work I'll use regex find and replaces to be more precise.

Issue: FEI-4957

## Test plan:
- yarn lint
- yarn test

### CODE REVIEW

**Review by jeresig — APPROVED** [review]
Awesome! Can you update the test plan with the command(s) that you used to make this change?

File: .eslintrc.js [hunk 1]
```diff
@@ -74,7 +74,7 @@ module.exports = {
         "import/no-named-default": "error",
         "import/extensions": [
             "error",
-            "always",
+            "never",
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply .eslintrc.js:77]: That's a good question.  https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/extensions.md doesn't say it is, but I could try using `--fix` on the next repo to see if it is.

File: .flowconfig [hunk 2]
```diff
@@ -13,7 +13,6 @@ module.use_strict=true
 
 [lints]
 deprecated-type=error
-deprecated-utility=error
```
[the remark this answers is not in the document]
Comment (kevinbarabash) [reply .flowconfig:16]: Flow complained that this is no longer a valid error.  I'm not sure why this wasn't caught earlier, but I've seen this happen in other repos too.

**Comment by MiguelCastillo** [comment]
this makes me sad!  I really got used to using extensions and learned to appreciate the extra readability it gives folks. oh well.

**Comment by somewhatabstract** [comment]
> TypeScript doesn't work with file extensions in imports.

I presume this means that we now have the ambiguity of not knowing if `./some-thing` is importing a `.ts`, `.tsx`, `./some-thing/index.ts`, or `./some-thing/index.tsx` just from reading the imports. Is that right?

I don't envisage that causing any issues (pretty sure the `./some-thing.js` could be `./some-thing.js/index.js` anyway, though that wouldn't happen), just wanted to understand any implications for my own personal edification.

**Review by somewhatabstract — APPROVED** [review]
This looks good to me. Thanks for documenting methodology and the reasoning behind this. I think, especially in these early changes, being deliberate about documenting the methodology in the PR description, including what went well and what didn't, will be really helpful as we do this migration.

**Pull request merged**